### PR TITLE
Add datasource for teams

### PR DESCRIPTION
**Pull request by andredupreez**
This is for an implementation of the missing team terraform data source, which is required for referencing existing team resources for the purpose of referencing the team ID during folder permissions configuration.

### CODE REVIEW

File: grafana/data_source_team_test.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+func TestAccDatasourceTeam(t *testing.T) {
+	CheckOSSTestsEnabled(t)
+
+	var team gapi.Team
+	checks := []resource.TestCheckFunc{
+		testAccTeamCheckExists("grafana_team.test", &team),
+		resource.TestCheckResourceAttr(
+			"data.grafana_team.from_name", "name", "test-team",
+		),
```
Comment (julienduchesne) [inline grafana/data_source_team_test.go:18]:
```suggestion
		resource.TestCheckResourceAttr(
			"data.grafana_team.from_name", "name", "test-team",
		),
		resource.TestMatchResourceAttr("data.grafana_team.from_name", "id", idRegexp),
```

Check the id as well

**Review by julienduchesne — COMMENTED** [review]
Awesome. Just needs a `go generate ./...` and one test nit!

File: grafana/data_source_team_test.go (hunk 1 above)
Comment (andredupreez) [reply grafana/data_source_team_test.go:18]: Ok thanks, made the changes. Let me know if there is anything else

**Review by julienduchesne — APPROVED** [review]

**Comment by julienduchesne** [comment]
Thanks for your contribution!

**Pull request merged**